File: flutter/rsvp/lib/Models/event.dart
import 'Response.dart';

class Event {
  String _link;
  String _name;
  DateTime _dateTime;
  String _description;
  String _creator;
  String _location;
  int _minNum;
  bool _status;
  List<Response> _allResponses;

  Map toJson() => { // TODO: Add _allResponse
    'link': _link,
    'name': _name,
    'date': dateTime.toString(),
    'description': _description,
    'creator': _creator,
    'location': _location,
    'minNum': minNum.toString(),
    'status': status.toString(),
  };

  factory Event.fromJson(dynamic json) { // TODO: Add _allResponse
    return Event(json['link'] as String, json['name'] as String, json['creator'] as String,
        json['description'] as String, DateTime.parse(json['date'] as String),
        json['location'] as String, json['minNum'] as int, json['status'] as bool);
  }

  Event( // TODO: Add _allResponse
      String link,
      String name,
      String creator,
      String description,
      DateTime dateTime,
      String location,
      int minNum,
      bool status) {
    _link = link;
    _name = name;
    _creator = creator;
    _description = description;
    _dateTime = dateTime;
    _location = location;
    _minNum = minNum;
    _status = status;
    _allResponses = [];
  }

  String get link => _link;

  String get name => _name;

  int get minNum => _minNum;

  String get creator => _creator;

  DateTime get dateTime => _dateTime;

  String get location => _location;

  bool get status => _status;

  String get description => _description;

  List<Response> get allResponses => _allResponses;

  void addNewResponse(Response response) {
    _allResponses.add(response);
  }

//  set allResponses(List<Response> value) {
//    _allResponses = value;
//  }

  set description(String value) {
    _description = value;
  }

  set link(String value) {
    _link = value;
  }

  set name(String value) {
    _name = value;
  }

  set minNum(int value) {
    _minNum = value;
  }

  set creator(String value) {
    _creator = value;
  }

  set dateTime(DateTime value) {
    _dateTime = value;
  }

  set location(String value) {
    _location = value;
  }

  set status(bool value) {
    _status = value;
  }
}

File: service/src/main/java/edu/seattleu/rsvp/service/dtos/Event.java
package edu.seattleu.rsvp.service.dtos;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import edu.seattleu.rsvp.service.entities.EventEntity;

public class Event implements Serializable
{
    private String link;
    private String name;
    private Date dateTime;
    private String description;
    private String creator;
    private String location;
    private int minNum;
    private boolean status;
    private List<Response> responses;

    public Event()
    {
    }

    public Event(String link, String name, Date dateTime, String description, String creator,
                 String location, int minNum, boolean status, List<Response> responses)
    {
        this.link = link;
        this.name = name;
        this.dateTime = dateTime;
        this.description = description;
        this.creator = creator;
        this.location = location;
        this.minNum = minNum;
        this.status = status;
        this.responses = responses;
    }

    public static Event createFromEntity(EventEntity eventEntity, List<Response> allResponses)
    {
        return new Event(eventEntity.getLink(), eventEntity.getName(), eventEntity.getDateTime(),
                eventEntity.getDescription(), eventEntity.getCreator(), eventEntity.getLocation(),
                eventEntity.getMinNum(), eventEntity.isHappening(), allResponses);
    }

    public String getLink()
    {
        return link;
    }

    public String getName()
    {
        return name;
    }

    public Date getDateTime()
    {
        return dateTime;
    }

    public String getDescription()
    {
        return description;
    }

    public String getCreator()
    {
        return creator;
    }

    public String getLocation()
    {
        return location;
    }

    public int getMinNum()
    {
        return minNum;
    }

    public boolean isHappening()
    {
        return status;
    }

    public List<Response> getResponses()
    {
        return responses;
    }

    public void setLink(String link)
    {
        this.link = link;
    }

    public void addResponse(Response aNewResponse)
    {
        responses.add(aNewResponse);
    }
}